File: java/AnimalBase.java
public abstract class AnimalBase implements IAnimal {
    protected String speciesName;
    protected String name;

    public AnimalBase(String name, String speciesName) {
        this.name = name;
        this.speciesName = speciesName;
    }

    @Override
    public String getName() { return name; }

    @Override
    public String getSpeciesName() { return speciesName; }

    @Override
    public String makeSound() { return "Sound made by Animal: "; }
}

File: python/animal_base.py
from ianimal import IAnimal

class AnimalBase(IAnimal):
    def __init__(self, species_name: str, name: str):
        self.species_name = species_name
        self.name = name
    def get_name(self):         return self.name
    def get_species_name(self): return self.species_name
    def make_sound(self):       return "Sound made by Animal: "
